AdventOfCode2023/Day7.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AdventOfCode2023
+{
+    internal class Day7
+    {
+        public static long Day7Main(string[] inputArray)
+        {
+            var hands = new List<Hand>();
+            long totalSum = 0;
+
+            foreach (var input in inputArray)
+                hands.Add(new Hand(input.Split(' ')[0], Convert.ToInt32(input.Split(' ')[1])));
+
+            hands.Sort();
+
+            for (var i = 0; i < hands.Count; i++)
+                totalSum += (i + 1) * hands[i].bid;
+
+            return totalSum;
+        }
+
+        class Hand : IComparable<Hand>
```

Comment:
It looks like you have a class within a class. Classes should be outside of other classes. Technically they should be different files (they can have the same namespace if you want your using statement to work for both classes)